### ISSUE TITLE
Selected Port Polling causes counter32 OID to be used for tunnel interface

### ISSUE THREAD

**Issue by cardonja** [librenms/librenms#16434]
### The problem

When Selected Port Polling is enabled, this condition  (includes/polling/ports.inc.php:270):

`                    if (is_numeric($port_stats[$ifIndex]['ifHighSpeed']) && $port_stats[$ifIndex]['ifHighSpeed'] > 0) {`
`                         $full_oids = array_merge($hc_oids, $shared_oids);`
`                     } else {`
`                         $full_oids = array_merge($nonhc_oids, $shared_oids);`
`                     }`

triggers the use of counter32 OID's to be used for  logical interfaces on devices that adhere to RFC1573.

RFC1573 states (p. 38):

>    ifHighSpeed OBJECT-TYPE
       SYNTAX      Gauge32
       MAX-ACCESS  read-only
       STATUS      current
       DESCRIPTION
               "An estimate of the interface's current bandwidth in
               units of 1,000,000 bits per second.  If this object
               reports a value of 'n' then the speed of the interface
               is somewhere in the range of 'n-500,000' to
               'n+499,999'.  For interfaces which do not vary in
               bandwidth or for those where no accurate estimation
               can be made, this object should contain the nominal
               bandwidth.  For a sub-layer which has no concept of
               bandwidth, this object should be zero."

In the aforementioned condition however, a zero value leads to nonhc_oids being used.

On high-speed tunnel interfaces, these OID's suffer from rollovers occuring within a poller-interval, causing faulty graphs.
Since the presence of the IfHighSpeed OID implies the availability of 64-bit counters, removing the 'larger than zero'-test from  the condition is all that is needed for LibreNMS to start using the 64-bit counters.

### Output of ./validate.php

```txt
librenms@nms:~$ ./validate.php
===========================================
Component | Version
--------- | -------
LibreNMS  | 24.8.0-84-gd23ecf551 (2024-09-24T23:18:55+02:00)
DB Schema | 2024_08_27_182000_ports_statistics_table_rev_length (299)
PHP       | 8.2.23
Python    | 3.9.2
Database  | MariaDB 10.5.26-MariaDB-0+deb11u2
RRDTool   | 1.7.2
SNMP      | 5.9
===========================================

[OK]    Composer Version: 2.7.9
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQL and PHP time match
[OK]    Distributed Polling setting is enabled globally
[OK]    Connected to rrdcached
[OK]    Active pollers found
[OK]    Dispatcher Service is enabled
[OK]    Locks are functional
[OK]    Python wrapper cron entry is not present
[OK]    Redis is functional
[OK]    rrdtool version ok
[OK]    Connected to rrdcached
[WARN]  Your local git contains modified files, this could prevent automatic updates.
	[FIX]:
	You can fix this with ./scripts/github-remove
	Modified Files:
	 includes/polling/ports.inc.php
```

**Comment by PipoCanaja** [comment]
Hi @cardonja 
Thanx for your issue. It makes sense. I don't have any historical reason on why it was done like so. 
Could you open a Pull Request with the fix, and we'll see how the CI tests will do with it.
Thanx

**Comment by laf** [comment]
I'm expecting that check is in due to poor snmp implementations

**Comment by PipoCanaja** [comment]
> I'm expecting that check is in due to poor snmp implementations

As always :) Let's try to follow the RFC and handle the broken ones instead of breaking a correct implementation. If we have enough test-data, this should appear in the PullRequest CI.

**Comment by murrant** [comment]
This is a land mine :)  There are a lot of buggy implementations out there and this is trying to have the best outcome.

I'm open to proposals that can detect when HC counters should be used without fetching the entire HC table.   Likely, you will fix some and break others, but it is still worth trying.

**Comment by mourala-debug** [comment]
Any news about this issue? Couldn't find the pull request or any other related.

**Comment by murrant** [comment]
@mourala-debug nope, no one has figured out a way to deal with both buggy snmp implementations and ports that return 0 for ifHighSpeed.

**Comment by mourala-debug** [comment]
> @mourala-debug nope, no one has figured out a way to deal with both buggy snmp implementations and ports that return 0 for ifHighSpeed.

Thanks a lot, for the answer and for keeping alive such a great and lovely project.

**Comment by murrant** [comment]
@mourala-debug best case option for individuals is to report bad snmp behavior to vendors.

**Comment by laf** [comment]
Closing due to lack of updates or pull request
